tests/tools/oecertdump/host/host.cpp

```diff
@@ -0,0 +1,323 @@
+    if ((result != OE_OK) || (ecall_result != OE_OK))
+    {
+        log("Failed to create certificate. Enclave: %s, Host: %s\n",
+            oe_result_str(ecall_result),
+            oe_result_str(result));
+
+        goto exit;
+    }
+    else
+    {
+        output_certificate(cert, cert_size);
+        validate_certificate(cert, cert_size);
+    }
+    if (cert)
+    {
+        free(cert);
+        cert = NULL;
+    }
+    cert_size = 0;
+
+    // RSA Key
+    result = get_tls_cert_signed_with_rsa_key(
+        enclave, &ecall_result, &cert, &cert_size);
+    if ((result != OE_OK) || (ecall_result != OE_OK))
+    {
+        log("Failed to create certificate. Enclave: %s, Host: %s\n",
+            oe_result_str(ecall_result),
+            oe_result_str(result));
+
+        goto exit;
+    }
+    else
+    {
+        output_certificate(cert, cert_size);
+        validate_certificate(cert, cert_size);
+    }
+
+exit:
+    // deallcate resources
+    if (cert)
+        free(cert);
+
+    return result;
+}
+
+static void _display_help(const char* cmd)
+{
+    printf("Usage: %s ENCLAVE_PATH Options\n", cmd);
+    printf("\tOptions:\n");
+    printf("\t%s : output filename.\n", INPUT_PARAM_OPTION_OUT_FILE);
+}
+
+static int _parse_args(int argc, const char* argv[])
+{
+    if (argc < 1)
+    {
+        _display_help(argv[0]);
+        return 1;
+    }
+
+    // clear params memory
+    memset(&_params, 0, sizeof(_params));
+
+    int i = 1; // current index
+    // save
+    _params.enclave_filename = argv[i++];
+    _params.out_filename = DEFAULT_OUTPUTFILE;
+
+    // Verify enclave file is valid
+    FILE* fp = fopen(_params.enclave_filename, "rb");
+    if (!fp)
+    {
+        printf("Failed to find file: %s\n", _params.enclave_filename);
+        return 1;
+    }
+    else
+        fclose(fp);
+
+    while (i < argc)
+    {
+        if (strcmp(INPUT_PARAM_OPTION_OUT_FILE, argv[i]) == 0)
+        {
+            if (argc >= i + 1)
+            {
+                _params.out_filename = argv[i + 1];
+                i += 2;
+            }
+            else
+            {
+                printf(
+                    "%s has invalid number of parameters.\n",
+                    INPUT_PARAM_OPTION_OUT_FILE);
+                _display_help(argv[0]);
+                return 1;
+            }
+        }
+        else if (strcmp(INPUT_PARAM_USAGE, argv[i]) == 0)
+        {
+            _display_help(argv[0]);
+            return 1;
+        }
+        else
+        {
+            printf("Invalid option: %s\n", argv[i]);
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+static oe_result_t _process_params(oe_enclave_t* enclave)
+{
+    oe_result_t result = OE_FAILURE;
+
+    result = gen_report(enclave);
+    if (result != OE_OK)
+        return result;
+
+    result = _gen_cert(enclave);
+
+    return result;
+}
+
+#endif // OE_LINK_SGX_DCAP_QL
+
+int main(int argc, const char* argv[])
+{
+    int ret = 0;
+
+#ifdef OE_LINK_SGX_DCAP_QL
```

Comment:
question: who will define OE_LINK_SGX_DCAP_QL?

Reply:
OE_LINK_SGX_DCAP_QL is publicly defined in other CMakeLists.txt in the build chain (enclave/core/CMakeLists.txt and host/CMaktLists.txt).
It is also added to the our host/CMakeLists.txt as PRIVATE.